src/pip/_internal/req/req_install.py

```diff
@@ -241,7 +246,11 @@ def from_path(self):
         """
         if self.req is None:
             return None
-        s = str(self.req)
+        s = self.req.name
+        if self.extras:
+            s += '[' + ','.join(sorted(self.extras)) + ']'
```

Comment:
This would be better as `s += '[%s]' % ','.join(sorted(self.extras))`, and it improves performance.